### PR TITLE
Check that indices exist in healthcheck

### PR DESCRIPTION
**Pull request by sully90**
### What

Check that indices exist in healthcheck

### How to review

Code review and check tests pass

### Who can review

Anyone but me

### CODE REVIEW

File: dp_conceptual_search/api/healthcheck/routes.py [hunk 1]
```diff
@@ -37,7 +38,22 @@
...
+
+        # Check indices exist
+        indices = "{ons},{departments}".format(ons=SEARCH_CONFIG.search_index,
+                                               departments=SEARCH_CONFIG.departments_search_index)
+        indicies_exist = client.indices.exists(indices)
+        if isawaitable(indicies_exist):
+            indicies_exist = await indicies_exist
+
+        if indicies_exist:
+            return json(request, health, code)
```
Comment (janderson2) [inline dp_conceptual_search/api/healthcheck/routes.py]: Change response body as discussed.  i.e.:
```
# Healthy
{
  "elasticsearch": "OK"
}

# Cluster unhealthy:
{
  "elasticsearch": ("unavailable"|"indices not found"|"cluster status red")
}
```

File: swagger.yaml [hunk 2]
```diff
@@ -200,6 +219,34 @@ paths:
...
+          description: "Page size"
+          type: integer
+          required: false
+      summary: "Query for recommended content"
+      description: "Executes a query which returns content similar to the page at the given path. If the _ga cookie is specified, then the corresponding users session vector is also used."
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/SearchResponse'
```
Comment (janderson2) [inline swagger.yaml]: Does this endpoint also return `SearchResponse`?

File: swagger.yaml [hunk 3]
```diff
@@ -144,16 +116,101 @@ paths:
...
+        - in: query
+          name: sort_by
+          required: false
+          type: string
+          enum:
+            - first_letter
+            - title
+            - relevance
+            - release_date
+            - release_date_asc
```
Comment (janderson2) [inline swagger.yaml]: Is `release_date` the release date descending? If so would it be more consistent to put `release_date_desc`?

File: swagger.yaml [hunk 4]
```diff
@@ -144,16 +116,101 @@ paths:
...
+        - in: query
```
Comment (janderson2) [inline swagger.yaml]: It would probably be good to have a `description`.

File: swagger.yaml [hunk 5]
```diff
@@ -144,16 +116,101 @@ paths:
...
+          required: true
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/SearchResponse'
+        400:
+          description: Query term not specified
+        500:
+          description: Internal server error
```
Comment (janderson2) [inline swagger.yaml]: Does the conceptual search endpoint have the reworked `filters` logic? If so it needs to be updated in the spec.

Also, conceptual search still has the endpoint templated.

File: swagger.yaml [hunk 6]
```diff
@@ -144,16 +116,101 @@ paths:
...
           type: integer
           required: false
-          minimum: 1
+        - in: query
+          name: sort_by
+          required: false
+          type: string
+          enum:
+            - first_letter
+            - title
```
[the remark this answers is not in the document]
Comment (sully90) [reply swagger.yaml]: Added description

File: swagger.yaml (hunk 3 above)
Comment (sully90) [reply swagger.yaml]: Yes, it would be 👍 I totally support you in your decision to do this.

File: swagger.yaml (hunk 5 above)
Comment (sully90) [reply swagger.yaml]: Updated

**Pull request merged**